### DISCUSSION TITLE
Restoring Deck instance after WebGL context lost

### DISCUSSION THREAD

**Discussion by jcarenza** [visgl/deck.gl#7841] [Q&A] [answered]
I've been getting a lot of `Error: WebGL context is lost` errors after a tab/screen wakes up from sleep. I am able to capture this event in the `onError` handler, but not sure what the best way to restore the app is? I've tried `redraw()`, forcing a remount by incrementing a `key` property...but no luck.

any suggestions or examples for doing this in the right way?

**Comment by Pessimistress** [comment, marked as answer]
Once the context is lost, there is no way to "restore" the purged WebGL resources. You should recreate the Deck/DeckGL instance and all layers that it renders. You might want to save the last view state so that the camera of the new instance will start where the old one left off.
